Back button - pass full sentence

marie-j opened the pull request:
### Requirements

* Filling out the template is required. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* All new code requires tests to ensure against regressions

##  allow to pass full translated sentence instead of leaving the concatenation to component


### Description of the Change

Instead of concatenating the "back to" and "previous page" by default allow to pass full sentence

### Benefits

This will work in every language we want 
e.g: French

### Possible Drawbacks

None

### Applicable Issues

e.g: in french it would be 'Retour à la "page précédente"' but we will also have 'Retour à la "Lorem Ipsum"' if we keep it that way

antleblanc reviewed (dismissed):
:pray: Thank you @marie-j!

Changes looks good to me. It will be part of the next major release.

antleblanc commented:
As we plan to introduce a BREAKING CHANGE, maybe we can consider taking into account  following **Deprecated** notice.

```sh
$ git grep -i "deprecated" -- './packages/*' ':!*.md'
packages/base/core/src/less/icons.less:// Deprecated: Support for renamed/duplicate classnames below
packages/base/core/src/less/icons.less:@oui-icon-deprecated:
packages/base/core/src/less/icons.less:#oui > .icon-content(@oui-icon-deprecated);
packages/components/badge/src/less/badge.less:.oui-status, /* Deprecated: Support for old classname */
packages/components/button/src/less/button.less:  .oui-icon-external_link, /* Deprecated: Support for old classname */
packages/components/button/src/less/button.less:.oui-button_full-width, /* Deprecated: Support for old classname */
packages/components/button/src/less/button.less:  &.oui-button_icon-right .oui-icon-external_link, /* Deprecated: Support for old classname */
packages/components/button/src/less/button.less:  .oui-icon-external_link, /* Deprecated: Support for old classname */
packages/components/button/src/less/button.less:.oui-button_small-width, /* Deprecated: Support for old classname */
packages/components/button/src/less/button.less:.oui-button_large-height, /* Deprecated: Support for old classname */
packages/components/button/src/less/button.less:  &.oui-button_small-width, /* Deprecated: Support for old classname */
packages/components/button/src/less/button.less:  &.oui-button_large-height, /* Deprecated: Support for old classname */
packages/components/button/src/less/button.less:  &.oui-button_small-width, /* Deprecated: Support for old classname */
packages/components/button/src/less/button.less:  &.oui-button_large-height, /* Deprecated: Support for old classname */
packages/components/button/src/less/button.less:  &.oui-button_large-height, /* Deprecated: Support for old classname */
packages/components/button/src/less/button.less:  &.oui-button_small-width, /* Deprecated: Support for old classname */
packages/components/button/src/less/button.less:  &.oui-button_large-height, /* Deprecated: Support for old classname */
packages/components/checkbox/src/less/checkbox-thumbnail.less:.oui-checkbox_thumbnail-light, /* Deprecated: Support for old classname */
packages/components/chips/src/less/chips.less:  &_full-width, /* Deprecated: Support for old classname */
packages/components/datagrid/src/less/_mixins.less:      &_sortable-asc, /* Deprecated: Support for old classname */
packages/components/datagrid/src/less/_mixins.less:      &_sortable-desc, /* Deprecated: Support for old classname */
packages/components/datagrid/src/less/_mixins.less:      &_sortable-asc, /* Deprecated: Support for old classname */
packages/components/datagrid/src/less/_mixins.less:      &_sortable-desc, /* Deprecated: Support for old classname */
packages/components/navbar/src/less/navbar.less:      &.oui-icon-info_circle, /* Deprecated: Support for old classname */
packages/components/navbar/src/less/navbar.less:      &.oui-icon-success_circle, /* Deprecated: Support for old classname */
packages/components/navbar/src/less/navbar.less:      &.oui-icon-warning_circle, /* Deprecated: Support for old classname */
packages/components/navbar/src/less/navbar.less:      &.oui-icon-error_circle, /* Deprecated: Support for old classname */
packages/components/radio/src/less/radio-thumbnail.less:.oui-radio_thumbnail-light, /* Deprecated: Support for old classname */
packages/components/tile/src/less/tile.less:      .oui-icon-external_link, /* Deprecated: Support for old classname */
```

Or another option is that we can keep both `backTo` and `previousPage` and make them as deprecated.
And handle to *translation issue* with a new attribut/binding.

Please let me know what do you think.

marie-j commented:
> As we plan to introduce a BREAKING CHANGE, maybe we can consider taking into account following **Deprecated** notice.
> 
> ```shell
> $ git grep -i "deprecated" -- './packages/*' ':!*.md'
> packages/base/core/src/less/icons.less:// Deprecated: Support for renamed/duplicate classnames below
> packages/base/core/src/less/icons.less:@oui-icon-deprecated:
> packages/base/core/src/less/icons.less:#oui > .icon-content(@oui-icon-deprecated);
> packages/components/badge/src/less/badge.less:.oui-status, /* Deprecated: Support for old classname */
> packages/components/button/src/less/button.less:  .oui-icon-external_link, /* Deprecated: Support for old classname */
> packages/components/button/src/less/button.less:.oui-button_full-width, /* Deprecated: Support for old classname */
> packages/components/button/src/less/button.less:  &.oui-button_icon-right .oui-icon-external_link, /* Deprecated: Support for old classname */
> packages/components/button/src/less/button.less:  .oui-icon-external_link, /* Deprecated: Support for old classname */
> packages/components/button/src/less/button.less:.oui-button_small-width, /* Deprecated: Support for old classname */
> packages/components/button/src/less/button.less:.oui-button_large-height, /* Deprecated: Support for old classname */
> packages/components/button/src/less/button.less:  &.oui-button_small-width, /* Deprecated: Support for old classname */
> packages/components/button/src/less/button.less:  &.oui-button_large-height, /* Deprecated: Support for old classname */
> packages/components/button/src/less/button.less:  &.oui-button_small-width, /* Deprecated: Support for old classname */
> packages/components/button/src/less/button.less:  &.oui-button_large-height, /* Deprecated: Support for old classname */
> packages/components/button/src/less/button.less:  &.oui-button_large-height, /* Deprecated: Support for old classname */
> packages/components/button/src/less/button.less:  &.oui-button_small-width, /* Deprecated: Support for old classname */
> packages/components/button/src/less/button.less:  &.oui-button_large-height, /* Deprecated: Support for old classname */
> packages/components/checkbox/src/less/checkbox-thumbnail.less:.oui-checkbox_thumbnail-light, /* Deprecated: Support for old classname */
> packages/components/chips/src/less/chips.less:  &_full-width, /* Deprecated: Support for old classname */
> packages/components/datagrid/src/less/_mixins.less:      &_sortable-asc, /* Deprecated: Support for old classname */
> packages/components/datagrid/src/less/_mixins.less:      &_sortable-desc, /* Deprecated: Support for old classname */
> packages/components/datagrid/src/less/_mixins.less:      &_sortable-asc, /* Deprecated: Support for old classname */
> packages/components/datagrid/src/less/_mixins.less:      &_sortable-desc, /* Deprecated: Support for old classname */
> packages/components/navbar/src/less/navbar.less:      &.oui-icon-info_circle, /* Deprecated: Support for old classname */
> packages/components/navbar/src/less/navbar.less:      &.oui-icon-success_circle, /* Deprecated: Support for old classname */
> packages/components/navbar/src/less/navbar.less:      &.oui-icon-warning_circle, /* Deprecated: Support for old classname */
> packages/components/navbar/src/less/navbar.less:      &.oui-icon-error_circle, /* Deprecated: Support for old classname */
> packages/components/radio/src/less/radio-thumbnail.less:.oui-radio_thumbnail-light, /* Deprecated: Support for old classname */
> packages/components/tile/src/less/tile.less:      .oui-icon-external_link, /* Deprecated: Support for old classname */
> ```
> 
> Or another option is that we can keep both `backTo` and `previousPage` and make them as deprecated.
> And handle to _translation issue_ with a new attribut/binding.
> 
> Please let me know what do you think.

@antleblanc Removing deprecated elements could be great but it won't be the same job to make apps compatible given the diff introduced

Moreover, making bindings deprecated could be a solution but it means that we can end up with lots of deprecate binding by fear of having a breaking change, I find it confusing and not great for Developer experience

marie-j committed:
fix(back-button): allow to pass full translations

The current concatenation with quotes doesn't work with all languages
e.g: in french it would be 'Retour à la "page précédente"' but we will also have 'Retour à la "Lorem Ipsum"'

Signed-off-by: Marie JONES <marie.jones@corp.ovh.com>

marie-j commented:
@antleblanc As spoken and given the context the previous way as been deprecated instead of being removed
